itest/performance/config.go

```diff
@@ -0,0 +1,136 @@
+package perf_test
+
+import (
+	"github.com/jessevdk/go-flags"
+)
+
+const (
+	// defaultConfigPath is the default path of the configuration file.
+	defaultConfigPath = "perf-test.conf"
+)
+
+// User defines the config options for a user in the network.
+type User struct {
+	Tapd *TapConfig `group:"tapd"  namespace:"tapd"`
+}
+
+// TapConfig are the main parameters needed for identifying and creating a grpc
+// client to a tapd subsystem.
+type TapConfig struct {
+	Name string `long:"name" description:"the name of the tapd instance"`
+	Host string `long:"host" description:"the host to connect to"`
+	Port int    `long:"port" description:"the port to connect to"`
+
+	TLSPath string `long:"tlspath" description:"Path to tapd's TLS certificate, leave empty if TLS is disabled"`
+	MacPath string `long:"macpath" description:"Path to tapd's macaroon file"`
+}
+
+// BitcoinConfig defines exported config options for the connection to the
+// btcd/bitcoind backend.
+type BitcoinConfig struct {
+	Host     string `long:"host" description:"bitcoind/btcd instance address"`
+	Port     int    `long:"port" description:"bitcoind/btcd instance port"`
+	User     string `long:"user" description:"bitcoind/btcd user name"`
+	Password string `long:"password" description:"bitcoind/btcd password"`
+	TLSPath  string `long:"tlspath" description:"Path to btcd's TLS certificate, if TLS is enabled"`
+}
+
+// Config holds the main configuration for the performance testing binary.
+type Config struct {
+	// TestCases is a comma separated list of test cases that will be
+	// executed.
+	TestCases string `long:"test-case" description:"the test case that will be executed"`
```

Comment:
I think this can be a `[]byte`, then it'll pick it up as a slice. So if we want cases `A`, `B`, and `C`, the binary can use the args: `--testcase=A --testcase=B --testcase=B`. This way, any upstream templating (in helm or w/e) doesn't need to handle the CSV format.

Reply:
You mean `[]string`, right? Otherwise it'll still be a single string.

Reply:
Err yeah, `[]string`.